src/cpp/rtps/security/SecurityPluginFactory.cpp

```diff
@@ -75,3 +76,20 @@ Cryptography* SecurityPluginFactory::create_cryptography_plugin(const PropertyPo
 
     return plugin;
 }
+
+Logging* SecurityPluginFactory::create_logging_plugin(const PropertyPolicy& property_policy)
+{
+    Logging* plugin = nullptr;
+    const std::string* logging_plugin_property = PropertyPolicyHelper::find_property(property_policy,
+            "dds.sec.log.plugin");
+
+    if(logging_plugin_property != nullptr)
+    {
+        if(logging_plugin_property->compare("builtin.DDS_LogTopic") == 0)
+        {
+            plugin = new LogTopic();
```

Comment:
It's not immediately obvious to me: what is responsible for deallocating this?

Reply:
The `SecurityManager` is, see [here](https://github.com/artivis/Fast-RTPS/blob/feature%2Fsecurity-logging/src/cpp/rtps/security/SecurityManager.cpp#L521).

Reply:
Ah thank you.